pantheon-bundle/src/main/java/com/redhat/pantheon/servlet/assembly/AssemblyVariantJsonServlet.java

```diff
@@ -201,7 +202,10 @@ protected boolean isValidResource(@Nonnull SlingHttpServletRequest request, @Non
                         .variants().get()
                         .canonicalVariant().get();
                 moduleMap.put("canonical_uuid", canonical.uuid().get());
-                moduleMap.put("title", page.title().get());
+                // Get the current title from module instead of using the stored value in the content node
+                // When module title is modified, the value stored in content node becomes stale.
+                // Sometime, the title stored in content node shows filename instead of title.
+                moduleMap.put("title", ServletHelper.getModuleTitleFromUuid(canonical));
```

Comment:
I see what you're doing here - it seems like your thought is "page.title() doesn't always give me the information I need, so let me get it from somewhere else."

Here's the thing though... page.title() is exactly what we should be using, so if that's broken, then we need to fix it.

page.title() is populated in AssemblyVersion.consumeTableOfContents(). It looks like it's coded to use the draft title metadata field, and if the draft doesn't exist, it falls back to filename.

ServletHelper.getModuleTitleFromUuid() has logic that is similar - it pulls the draft metadata title, then tries released, then falls back to blank.

Since that second case is where you want to go, I would suggest fixing consumeTableOfContents to call latest() instead of draft() in its current logic. I think that would strike a good balance. What do you think?